src/components/Form/Form.tsx

```diff
@@ -0,0 +1,133 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import * as _ from 'lodash'
+
+import { UIComponent, childrenExist, customPropTypes } from '../../lib'
+import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
+import {
+  ComponentEventHandler,
+  Extendable,
+  ReactChildren,
+  ShorthandValue,
+} from '../../../types/utils'
+import { Sizes } from '../../lib/enums'
+import Grid from '../Grid/Grid'
+import FormField from './FormField'
+
+export interface IFormProps {
+  action?: string
+  as?: any
+  children?: ReactChildren
+  className?: string
+  onSubmit?: ComponentEventHandler<IFormProps>
+  columns?: string | number
+  content?: ShorthandValue | ShorthandValue[]
+  rows?: string | number
+  size?: Sizes
+  loading?: boolean
+  styles?: ComponentPartStyle
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A Form displays a set of related user input fields in a structured way.
+ */
+class Form extends UIComponent<Extendable<IFormProps>, any> {
+  static create: Function
+
+  public static displayName = 'Form'
+
+  public static className = 'ui-form'
+
+  public static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /**
+     *  Button content for childrenApi
+     *  @docSiteIgnore
+     */
+    children: PropTypes.node,
+
+    /** Additional CSS class name(s) to apply.  */
+    className: PropTypes.string,
+
+    /** The columns of the grid with a space-separated list of values. The values represent the track size, and the space between them represents the grid line. */
+    columns: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
```

Comment:
I would rather consider to remove layout-related props from the Form's API for now, as am pretty sure that we would be able to address quite wide set of scenarios by
- introducing base 'vertical form' layout support as a default one
- ability to customize form styles for the client - not that with that it would be possible for the client to provide custom layout styles for each Form.Field element
- potentially add ability to introduce support for horizontal forms

Not saying that **_for sure we won't need Grid support for form's layout_** - the only thing that stating is that _for now_ we, probably, don't need it

Reply:
This seemed like the easier (for implementation and usage) layouting mechanism support out of the box for the form component. With this by default we are supporting the vertical, multi-column and horizontal layout. We might introduce this as a consequent PR too, after the initial implementation is merged. What are your thoughts on this? And, yes, I would definitelly use the Grid in the background anyway for aligning the elements